include/envoy/ssl/context_config.h

```diff
@@ -111,6 +112,16 @@ class ContextConfig {
    * @return The maximum TLS protocol version to negotiate.
    */
   virtual unsigned maxProtocolVersion() const PURE;
+
+  /**
+   * @return true of the config is valid.
+   */
+  virtual bool isValid() const PURE;
+
+  /**
+   * @return the DynamicSecretProvider object.
+   */
+  virtual Secret::DynamicSecretProvider* getDynamicSecretProvider() const PURE;
```

Comment:
can this return a const pointer?

Reply:
Please remove code that is not used in this PR. It's hard to review without context.